packages/eslint-plugin/docs/rules/no-duplicate-type-constituents.md

```diff
@@ -0,0 +1,49 @@
+---
+description: 'Disallow duplicate union/intersection type members.'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-duplicate-type-constituents** for documentation.
+
+Although TypeScript supports duplicate union and intersection member values, people usually expect members to have unique values within the same intersection and union. Duplicate values make the code redundant and generally reduce readability.
+
+## Rule Details
+
+This rule disallows duplicate union or intersection type members. It only checks duplication on the notation. Members with the same value but different names are not marked duplicates.
```

Comment:
[Docs] I don't think this is accurate anymore?

I also don't know that people would know what _"same value"_ means. Could you elaborate?
* We do check whether type equality ends up being the same
* But sometimes TypeScript treats things that are the same type as different ones:
    ```ts
    type WrapInValue<T> = { value: T };
    type T = { value: number } | WrapInValue<number>;
    ``` 

I'd suggest adding to the docs some explanation of how this rule determines whether two types are equivalent: that we check whether the syntax is the same, or if TypeScript treats them as the exact same type. This would both help explain edge cases to users and help explain why the code is doing _two_ equality checks.

Yet again I am wishing for https://github.com/microsoft/TypeScript/issues/9879 🥲 which would allow us to skip all this logic, and maybe even combine the rule with `no-redundant-type-constituents`...